src/Conversion/ONNXToTorch/NN/CommonUtils.cpp

```diff
@@ -1,62 +1,44 @@
 #include "CommonUtils.h"
 
-typedef struct dim_pads {
-  int dim_start;
-  int dim_end;
-} dim_pads;
-
 std::vector<Value>
 createPadsArrayAttribute(::mlir::ArrayAttr pads, Type ty, Location loc,
                          ConversionPatternRewriter &rewriter) {
-  // Reading the ONNX side pads values and store in the array.
+  // Read ONNX side pads values and store inside a vector
   std::vector<Value> translatepadsList;
-  if (!pads)
-    return translatepadsList;
-
-  bool is_symmetric = true;
-  for (unsigned int i = 0; i < pads.size(); i += 2) {
-    if (pads[i] != pads[i + 1]) {
-      is_symmetric = false;
-      break;
-    }
-  }
-
-  dim_pads dimArray[pads.size()];
-  if (is_symmetric) {
-    for (unsigned int i = 0; i < pads.size(); i += 2) {
-      auto pad_value =
-          (pads[i].dyn_cast<IntegerAttr>()).getValue().getZExtValue();
-      auto f0 = IntegerAttr::get(ty, pad_value);
-      Value p0v = rewriter.create<ConstantIntOp>(loc, f0);
-      translatepadsList.push_back(p0v);
+  if (!pads)  {
+    for (unsigned i = 0; i < 2; i++) {
+      Value zeroPaddingValue = rewriter.create<ConstantIntOp>(loc,
+          IntegerAttr::get(ty, 0));
+      translatepadsList.push_back(zeroPaddingValue);
     }
   } else {
-    int j = 0;
-    for (unsigned int i = 0; i < pads.size(); i++) {
-      dimArray[j].dim_start =
-          (pads[i].dyn_cast<IntegerAttr>()).getValue().getZExtValue();
-      i++;
-      dimArray[j].dim_end =
-          (pads[i].dyn_cast<IntegerAttr>()).getValue().getZExtValue();
-      j++;
-    }
-
-    // read the onnx pad values from array(dim_start values)
-    int k = 0;
-    for (unsigned int i = 0; i < pads.size(); i = i + 2) {
-      auto f0 = IntegerAttr::get(ty, (dimArray[k].dim_start));
-      Value p0v = rewriter.create<ConstantIntOp>(loc, f0);
-      translatepadsList.push_back(p0v);
-      k++;
-    }
+    // Determine if padding is symmetrical
+    // `onnx-mlir` padding has the following form
+    // (pad_dim1_start, pad_dim2_start, pad_dim1_end, pad_dim2_end)
+    bool is_symmetric = true;
+    if (pads[0] != pads[2] || pads[1] != pads[3])
+      is_symmetric = false;
 
-    // read the onnx pad values from array(dim_end values)
-    k = 0;
-    for (unsigned int i = 0; i < pads.size(); i = i + 2) {
-      auto f1 = IntegerAttr::get(ty, (dimArray[k].dim_end));
-      Value p1v = rewriter.create<ConstantIntOp>(loc, f1);
-      translatepadsList.push_back(p1v);
-      k++;
+    // Create appropriate padding vectors based on padding symmetry
+    if (is_symmetric) {
+      for (unsigned i = 0; i < pads.size(); i += 2) {
+        auto pad = (pads[i].dyn_cast<IntegerAttr>()).getValue().getZExtValue();
+        auto padAttr = IntegerAttr::get(ty, pad);
+        Value padValue = rewriter.create<ConstantIntOp>(loc, padAttr);
+        translatepadsList.push_back(padValue);
```

Comment:
There is code duplication here, translating an `IntegerAttr` to a `ConstantIntOp`. Please use a helper function.